src/app/services/product.service.ts

```diff
@@ -93,18 +85,17 @@ export class ProductService implements IproductService {
       let getProductListUrl: string = this.getUrl("/") + "products/" + productId;
       return this.http.get<GetProductDetailsResponse>(getProductListUrl, {
         headers: this.headers
-      }).pipe(retry(1),catchError(this.errorHandler));
+      }).pipe(retry(1), catchError(this.errorHandler));
```

Comment:
I would like to understand `getProducts` api as per contracts return you this response :
https://github.com/tavisca-sgogawale/OnlineRetailPortal/blob/develop/Contracts/v1.0/seller/response.json5
according to that:
````json
//Response data
{
	//[Mandatory] id of the product
	"id" : "P123",
	//[Mandatory] title of the product
	"title" : "Watch" ,
	//[Mandatory] description of the product
	"description" : "something something something",
	//[Mandatory] images of the product
	"heroImageUrl" : "www.example.com/imageName",
	//[Mandatory] 
	"price" : 
	{
.
.
.
.
.

````
there is no `product` property in the response nor the seller . 
Please get it validated from people worked on this api . As if seller field is null , then you are displaying the error page.(As per last discussion , we are only returning sellerId in the response.

Reply:
had a discussion with nikita.
the code so far is good to go.
the seller contract needs to be updated.